File: chapter-3-greedy/1이-될-때까지-book.cpp
#include <iostream>

using namespace std;

int solution(int N, int K){
    int count = 0;

    while(true) {
        // N이 K로 나누어 떨어지는 수가 될 때까지만 1씩 빼기
        int target = (N / K) * K;
        count += (N - target);
        N = target;
        // N이 K보다 작을 때 (더 이상 나눌 수 없을 때) 반복문 탈출
        if (N < K) break;
        // K로 나누기
        count += 1;
        N /= K;
    }

    // 마지막으로 남은 수에 다하여 1씩 빼기
    count += (N - 1);
    return count;
} 

int main(){
    int N, K;
    // N, K를 공백을 기준으로 구분하여 입력 받기
    cin >> N >> K;

    int result = solution(N,K);
    cout << "실행 결과: " << result << endl;
    return 0;
}

File: chapter-3-greedy/1이-될-때까지-book.py
import time

def solution(N,K):
    count = 0
    
    # N이 K 이상이라면 K로 계속 나누기
    while N >= K:
        # N이 K로 나누어 떨어지지 않는다면 N에서 1씩 빼기
        while N % K != 0:
            N -= 1
            count += 1
        # K로나누기
        N //= K
        count += 1

    # 마지막으로 남은 수에 대하여 1씩 빼기
    while N > 1 :
        N -= 1
        count += 1

    return count


# N이 K 이상이라면 K로 게속 나누기
N, K = map(int, input().split())

start_time = time.time()
result = solution(N,K)
end_time = time.time()

print('실행 결과:',result)
print('걸린 시간:',end_time - start_time)
